target_chains/ethereum/sdk/solidity/PythUtils.sol

```diff
@@ -12,25 +15,102 @@ library PythUtils {
     /// @dev Function will lose precision if targetDecimals is less than the Pyth price decimals.
     /// This method will truncate any digits that cannot be represented by the targetDecimals.
     /// e.g. If the price is 0.000123 and the targetDecimals is 2, the result will be 0
+    /// This function will overflow if the combined exponent(targetDecimals + expo) is greater than 77 or less than -77.
+    /// This function will also revert if prices combined with the targetDecimals are greater than 10 ** 77 or less than 10 ** -77.
     function convertToUint(
         int64 price,
         int32 expo,
         uint8 targetDecimals
     ) public pure returns (uint256) {
-        if (price < 0 || expo > 0 || expo < -255) {
-            revert();
+        if (price < 0) {
+            revert PythErrors.NegativeInputPrice();
         }
+        if (expo < -255) {
+            revert PythErrors.InvalidInputExpo();
+        }
+        int32 combinedExpo = int32(uint32(targetDecimals)) + expo;
```

Comment:
the name of this variable is odd -- this is an offset or multiplier for the result of some sort right? 

I would be careful with names like "expo" and "decimal" because one is the negative of the other. When I see this line, I go "wait you can't add a decimal and an expo and get an expo".